drivers/adc/adc_stm32.c

```diff
@@ -849,11 +849,23 @@ static int adc_stm32_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32G4X)
+	/* In the G4 model, once can access internal temperature on ADCs 1 and 5 */
+	if ((adc == (ADC_TypeDef *)ADC1_BASE && __LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR_ADC1) == channel_cfg->channel_id) ||
```

Comment:
Not sure what the preferred way is to handle these horrendously long conditionals.